gluonnlp/vocab.py

```diff
@@ -304,6 +304,11 @@ def set_embedding(self, *embeddings):
                 'The argument `embeddings` must be an instance or a list of instances of ' \
                 '`gluonnlp.embedding.TokenEmbedding`.'
 
+        assert all([embs.unknown_token for embs in embeddings]) or \
+            all([not embs.unknown_token for embs in embeddings]), \
```

Comment:
nit: a list is not needed here. a generator suffices.

Reply:
Lists will generate nicer error messages during tests if the assertion fails. But lets change it to a generator as it will improve performance.

Reply:
on second thought, it's probably the same as it's hard to imagine user feeding 1mm different embeddings here...